src/analyzer.ts

```diff
@@ -54,24 +52,34 @@ export interface Options {
  */
 export class Analyzer {
   private _parsers: Map<string, Parser<any>> = new Map<string, Parser<any>>([
-    ['html', new HtmlParser(this)],
+    ['html', new HtmlParser()],
     ['js', new JavaScriptParser()],
-    ['css', new CssParser(this)],
+    ['css', new CssParser()],
     ['json', new JsonParser()],
   ]);
 
   private _entityFinders = new Map<string, EntityFinder<any, any, any>[]>([
     [
       'html',
-      [new HtmlImportFinder(), new HtmlScriptFinder(), new HtmlStyleFinder()]
+      [
+        new HtmlImportFinder(), new HtmlScriptFinder(), new HtmlStyleFinder(),
+        new DomModuleFinder()
+      ]
+    ],
+    [
+      'js',
+      [
+        new PolymerElementFinder(), new BehaviorFinder(),
+        new VanillaElementFinder()
+      ]
     ],
-    ['js', [new PolymerElementFinder(), new BehaviorFinder()]],
   ]);
 
   private _loader: UrlLoader;
-  private _resolver: UrlResolver;
-  private _documents = new Map<string, Promise<Document<any, any>>>();
-  private _documentDescriptors = new Map<string, Promise<DocumentDescriptor>>();
+  private _resolver: UrlResolver|undefined;
```

Comment:
is `o: T|undefined` any different from `o?: T`? I thought not. Which should we prefer?